src/class-bc-asaas.php

```diff
@@ -0,0 +1,317 @@
+<?php
+// Don't load directly
+defined( 'WPINC' ) or die;
+
+class WC_Boleto_Control_Asaas {
+
+	/**
+	 * Static Singleton Holder
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 * @return self
+	 */
+	public static function instance(){
+		return self::$instance ? self::$instance : self::$instance = new self;
+	}
+
+	/**
+	 * API varibles stored in a single Object
+	 *
+	 * @var array $api {
+	 *     @type string     $key         License key for the API (PUE)
+	 *     @type string     $version     Which version of we are dealing with
+	 *     @type string     $domain      Domain in which the API lies
+	 *     @type string     $path        Path of the API on the domain above
+	 * }
+	 */
+	public $api = array(
+		'key' 		=> '6f1beb274b8b85e613d66a88109d7b6d72dee679d50d62c46c67c6a356ed4445',
+		'version' 	=> 'v2',
+		'domain' 	=> 'https://homolog.asaas.com/',
+		'path'		=> 'api/',
+	);
+
+	/**
+	 * Constructor!
+	 */
+	private function __construct() {
+
+		//turns api array into object
+		$this->api = (object) $this->api;
+	}
+
+	/**
+	 * Builds an endpoint URL
+	 *
+	 * @param string $endpoint  Endpoint for the Event Aggregator service
+	 * @param array  $data      Parameters to add to the URL
+	 *
+	 * @return string|WP_Error
+	 */
+	public function build_url( $endpoint, $data = array() ) {
+
+		return "{$this->api->domain}{$this->api->path}{$this->api->version}/{$endpoint}";
+	}
+
+	/**
+	 * Performs a GET request against the Event Aggregator service
+	 *
+	 * @param string $endpoint   Endpoint for the Event Aggregator service
+	 * @param array  $data       Parameters to send to the endpoint
+	 *
+	 * @return stdClass|WP_Error
+	 */
+	public function get( $endpoint, $data = array() ) {
+		$url = $this->build_url( $endpoint, $data );
+
+		// If we have an WP_Error we return it here
+		if ( is_wp_error( $url ) ) {
+			return $url;
+		}
+
+		$headers = array(
+			'access_token'  => $this->api->key,
+		);
+
+		var_dump($url);
+
+		$response = wp_remote_get( esc_url_raw( $url ), array( 'timeout' => 60, 'limit' => '50', 'headers' => $headers ) );
```

Comment:
Break the second argument into a variable `$args` of some sort, to make it more readable.